systemd/systemd.go

```diff
@@ -709,28 +766,45 @@ func (l Log) Time() (time.Time, error) {
 
 // Message of the Log, if any; otherwise, "-".
 func (l Log) Message() string {
-	if msg, ok := l["MESSAGE"]; ok {
-		return msg
+	// for MESSAGE, if there are multiple strings, just concatenate them with a
+	// newline to keep as much data from journald as possible
+	msg, err := l.parseLogRawMessageString("MESSAGE", func(stringSlice []string) (string, error) {
+		return strings.Join(stringSlice, "\n"), nil
+	})
+	if err != nil {
+		return "-"
```

Comment:
Should we DEBUG log the error here, in case we have a bug in the parser somehow and people are trying to find out why?

Reply:
I would like to import logger here to log a message, but we also have logging from the caller of this code in daemon, so I'm not sure we should be doing duplicated logging here, but I'm open to alternative proposals.

What I don't want to have happen is to have an application that issues lots of messages that we can't handle/transform here and flood snapd logs with useless messages about an issue with a message, because I imagine in an app finds itself sending a weird message, it has propensity to send _lots_ of those kinds of message.

Reply:
Perhaps we could use `- (error decoding original message)` to at least indicate a problem there.

Reply:
I modified the error message returned here to include the error but only if the error was something other than the MESSAGE key missing.